packages/core/src/base-axis-chart.ts

```diff
@@ -390,22 +393,25 @@ export class BaseAxisChart extends BaseChart {
 
 		const tickHeight = this.getLargestTickHeight(yAxisRef.selectAll(".tick"));
 
-		const yAxisTitleTranslate = {
-			x: - (yAxisRef.node().getBBox().height / 2),
-			y: - (tickHeight + Configuration.scales.tick.heightAddition) * 1.5
-		};
-
 		// Add y-axis title
 		if (this.innerWrap.select(".axis-label.y").nodes().length === 0 && this.options.scales.y.title) {
 			yAxisRef.append("text")
 				.attr("class", "y axis-label")
-				.attr("transform", `rotate(-90) translate(${yAxisTitleTranslate.x}, ${yAxisTitleTranslate.y})`)
-				.attr("text-align", "center")
+				.attr("alignment-baseline", "middle")
 				.text(this.options.scales.y.title);
-
-			this.svg.attr("transform", `translate(${this.innerWrap.select(".axis-label.y").node().getBBox().height}, 0)`);
 		}
 
+		const yAxisCenter = yAxisRef.node().getBBox().height / 2;
+		const yAxisLabelWidth = this.innerWrap.select(".axis-label.y").node().getBBox().width;
+
+		const yAxisTitleTranslate = {
+			x: - yAxisCenter + yAxisLabelWidth / 2,
+			y: - (tickHeight + Configuration.scales.tick.heightAddition) * 1.5
+		};
+
+		// Align y axis title on the y axis
+		this.innerWrap.select(".axis-label.y").attr("transform", `rotate(-90) translate(${yAxisTitleTranslate.x}, ${yAxisTitleTranslate.y})`);
+
```

Comment:
Shouldn't the above go back into the if statement

Reply:
Yes